pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,460 @@
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guacanalytics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/google/go-cmp/cmp"
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"go.uber.org/zap"
+
+	"github.com/99designs/gqlgen/graphql/handler"
+	"github.com/guacsec/guac/pkg/assembler/backends/inmem"
+	"github.com/guacsec/guac/pkg/assembler/graphql/generated"
+	"github.com/guacsec/guac/pkg/assembler/graphql/resolvers"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+const (
+	inmems = "inmem"
+)
+
+func ingestTestData(ctx context.Context, client graphql.Client) {
+	logger := logging.FromContext(ctx)
+
+	topns := "topns"
+	topns1 := "topns1"
+	topns2 := "topns2"
+	topns3 := "topns3"
+	ns := "ubuntu"
+	version := "1.19.0"
+	depns := "openssl.org"
+	depns2 := "openssl.org2"
+	opensslVersion := "3.0.3"
+
+	ingestDependencies := []struct {
+		topTopPkg1           model.PkgInputSpec
+		topTopPkg2           model.PkgInputSpec
+		topTopPkg3           model.PkgInputSpec
+		topPkg               model.PkgInputSpec
+		pkg                  model.PkgInputSpec
+		depPkg1              model.PkgInputSpec
+		depPkg2              model.PkgInputSpec
+		dependencyDirect     model.IsDependencyInputSpec
+		dependencyNotInRange model.IsDependencyInputSpec
+		dependencyIndirect   model.IsDependencyInputSpec
```

Comment:
how about `assembler.IngestPredicates` here? may streamline the test cases a bit more so its easier to add future test cases.

Reply:
Fixed (on new PR #969)